public/app/features/alerting/unified/components/settings/AlertmanagerConfig.tsx

```diff
@@ -0,0 +1,190 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import AutoSizer from 'react-virtualized-auto-sizer';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Alert, Button, CodeEditor, ConfirmModal, Stack, useStyles2 } from '@grafana/ui';
+
+import { reportFormErrors } from '../../Analytics';
+import { useAlertmanagerConfig } from '../../hooks/useAlertmanagerConfig';
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+import { GRAFANA_RULES_SOURCE_NAME, isVanillaPrometheusAlertManagerDataSource } from '../../utils/datasource';
+import { Spacer } from '../Spacer';
+
+export interface FormValues {
+  configJSON: string;
+}
+
+interface Props {
+  alertmanagerName: string;
+  onDismiss: () => void;
+  onSave: (dataSourceName: string, oldConfig: string, newConfig: string) => void;
+  onReset: (dataSourceName: string) => void;
+}
+
+export default function AlertmanagerConfig({ alertmanagerName, onDismiss, onSave, onReset }: Props): JSX.Element {
+  const { loading: isDeleting, error: deletingError } = useUnifiedAlertingSelector((state) => state.deleteAMConfig);
+  const { loading: isSaving, error: savingError } = useUnifiedAlertingSelector((state) => state.saveAMConfig);
+  const [showResetConfirmation, setShowResetConfirmation] = useState(false);
+
+  const readOnly = alertmanagerName ? isVanillaPrometheusAlertManagerDataSource(alertmanagerName) : false;
+  const isGrafanaManagedAlertmanager = alertmanagerName === GRAFANA_RULES_SOURCE_NAME;
+  const styles = useStyles2(getStyles);
+
+  const {
+    currentData: config,
+    error: loadingError,
+    isSuccess: isLoadingSuccessful,
+    isLoading: isLoadingConfig,
+  } = useAlertmanagerConfig(alertmanagerName);
+
+  const defaultValues = {
+    configJSON: config ? JSON.stringify(config, null, 2) : '',
+  };
+
+  const {
+    register,
+    setValue,
+    setError,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({
+    defaultValues,
+  });
+
+  // make sure we update the configJSON field when we receive a response from the `useAlertmanagerConfig` hook
+  useEffect(() => {
+    if (config) {
+      setValue('configJSON', JSON.stringify(config, null, 2));
+    }
+  }, [config, setValue]);
+
+  useEffect(() => {
+    if (savingError) {
+      setError('configJSON', { type: 'deps', message: savingError.message });
+    }
+  }, [savingError, setError]);
+
+  useEffect(() => {
+    if (deletingError) {
+      setError('configJSON', { type: 'deps', message: deletingError.message });
+    }
+  }, [deletingError, setError]);
+
+  // manually register the config field with validation
+  // @TODO sometimes the value doesn't get registered – find out why
```

Comment:
In the silences work I had to `reset` the form with values when received from the API - is that a similar thing happening here perhaps?